packages/dd-trace/src/config.js

```diff
@@ -813,14 +564,92 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
+    this._setBoolean(env, 'experimental.runtimeId', DD_TRACE_EXPERIMENTAL_RUNTIME_ID_ENABLED)
+    this._setString(env, 'experimental.exporter', DD_TRACE_EXPERIMENTAL_EXPORTER)
+    this._setBoolean(env, 'experimental.enableGetRumData', DD_TRACE_EXPERIMENTAL_GET_RUM_DATA_ENABLED)
+    this._setValue(env, 'sampler.rateLimit', DD_TRACE_RATE_LIMIT)
+    this._setBoolean(env, 'reportHostname', DD_TRACE_REPORT_HOSTNAME)
+    this._setString(env, 'scope', DD_TRACE_SCOPE)
+    this._setBoolean(env, 'profiling.enabled', coalesce(DD_EXPERIMENTAL_PROFILING_ENABLED, DD_PROFILING_ENABLED))
+    this._setBoolean(env, 'profiling.sourceMap', DD_PROFILING_SOURCE_MAP && !isFalse(DD_PROFILING_SOURCE_MAP))
+    this._setString(env, 'profiling.exporters', DD_PROFILING_EXPORTERS)
+    if (DD_TRACE_SPAN_ATTRIBUTE_SCHEMA) {
+      this._setString(env, 'spanAttributeSchema', validateNamingVersion(DD_TRACE_SPAN_ATTRIBUTE_SCHEMA))
+    }
+    this._setBoolean(env, 'spanRemoveIntegrationFromService', DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED)
+    if (DD_TRACE_PEER_SERVICE_MAPPING) {
+      this._setValue(env, 'peerServiceMapping', fromEntries(
+        process.env.DD_TRACE_PEER_SERVICE_MAPPING.split(',').map(x => x.trim().split(':'))
+      ))
+    }
+    this._setBoolean(env, 'startupLogs', DD_TRACE_STARTUP_LOGS)
+    this._setBoolean(env, 'telemetry.logCollection', coalesce(DD_TELEMETRY_LOG_COLLECTION_ENABLED, DD_IAST_ENABLED))
+    this._setValue(env, 'telemetry.heartbeatInterval', maybeInt(Math.floor(DD_TELEMETRY_HEARTBEAT_INTERVAL * 1000)))
+    this._setBoolean(env, 'telemetry.debug', DD_TELEMETRY_DEBUG)
+    this._setBoolean(env, 'telemetry.metrics', DD_TELEMETRY_METRICS_ENABLED)
+    this._setBoolean(env, 'telemetry.dependencyCollection', DD_TELEMETRY_DEPENDENCY_COLLECTION_ENABLED)
+    this._setString(env, 'protocolVersion', DD_TRACE_AGENT_PROTOCOL_VERSION)
+    this._setValue(env, 'tagsHeaderMaxLength', DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH)
+    this._setBoolean(env, 'appsec.enabled', DD_APPSEC_ENABLED && isTrue(DD_APPSEC_ENABLED))
+    this._setString(env, 'appsec.rules', DD_APPSEC_RULES)
+    if (DD_APPSEC_RULES) this._setBoolean(env, 'appsec.customRulesProvided', !!DD_APPSEC_RULES)
+    this._setValue(env, 'appsec.rateLimit', maybeInt(DD_APPSEC_TRACE_RATE_LIMIT))
+    this._setValue(env, 'appsec.wafTimeout', maybeInt(DD_APPSEC_WAF_TIMEOUT))
+    this._setString(env, 'appsec.obfuscatorKeyRegex', DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP)
+    this._setString(env, 'appsec.obfuscatorValueRegex', DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP)
+    this._setValue(env, 'appsec.blockedTemplateHtml', maybeFile(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML))
+    this._setValue(env, 'appsec.blockedTemplateJson', maybeFile(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON))
+    this._setBoolean(env, 'remoteConfig.enabled', coalesce(
+      DD_REMOTE_CONFIGURATION_ENABLED && isTrue(DD_REMOTE_CONFIGURATION_ENABLED),
+      !this._isInServerlessEnvirontment()
+    ))
```

Comment:
why do you need a coalesce here ? this could be achieved with the default value instead no ?

Reply:
Default is true. Say `DD_REMOTE_CONFIGURATION_ENABLED` is not set, and we are in serverless environment, then `remoteConfig.enabled` is false. `isInServerlessEnvironment` is calculated from env vars, so we can't really call it default

Reply:
Ok then why is it not in the calculated part of the config like would be `telemetry.enabled` which is basically the same principle ?

Reply:
_applyCalculated only handles entries that are calculated from a mixture of env vars and options. remoteConfig.enabled is entirely calculated from env vars. Since _merge relies on the general rule that options take precedence over env vars, we created _applyCalculated to handle mixings of the two. I'll make it clear in a comment. I have a [question](https://dd.slack.com/archives/C02LESJ9PQX/p1708538513698689) about this config value,

Reply:
From our discussion: `telemetry.enabled ` should instead be moved in the env part